java/core/src/test/org/apache/orc/impl/TestInStream.java

```diff
@@ -434,8 +437,30 @@ public void testCompressed() throws Exception {
       int x = in.read();
       assertEquals(i & 0xff, x);
     }
+
+    // Forward seek test
+    int seeksPerformed = 0;
+    for(int i=0; i < 1024; ++i) {
+      long pos = positions[i].getNext();
+      positions[i].reset();
+      ByteBuffer c = ((InStream.CompressedStream) in).compressed;
+      if (pos == ((InStream.CompressedStream) in).currentCompressedStart) {
+        // Should have the same ByteBuffer if no seek has taken place
+        in.seek(positions[i]);
+        assertEquals(c, ((InStream.CompressedStream) in).compressed);
+      } else {
+        seeksPerformed += 1;
+        in.seek(positions[i]);
+        assertNotEquals(c, ((InStream.CompressedStream) in).compressed);
+      }
+      positions[i].reset();
+      assertEquals(i & 0xff, in.read());
+    }
+    assertEquals(1, seeksPerformed);
```

Comment:
Why are we assuming 1 seek here? do we have 2 chunks total?
Shall we move this to its own test?

Reply:
Makes sense let me add a separate test for this